app/forms/keywords_form.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+require 'csv'
+
+class KeywordsForm
+  include ActiveModel::Model
+
+  validates_with KeywordsFormValidator
+
+  attr_reader :file, :user
+
+  def initialize(user)
+    @user = user
+  end
+
+  def save(csv_file)
+    @file = csv_file
+
+    return false if invalid?
+
+    begin
+      save_keywords_to_db
+    rescue ActiveRecord::ActiveRecordError
+      errors.add('Invalid File')
+    end
+
+    errors.empty?
+  end
+
+  private
+
+  def save_keywords_to_db
+    keyword_records = parse_keywords.map { |keyword| add_keyword_record(keyword) }
+
+    SaveKeywordsToDb.new(keyword_records).call
+  end
```

Comment:
From the previous comment, how about renaming and using the form attribute `keywords`

```suggestion
  def save_keywords
    keyword_records = keywords.map { |keyword| add_keyword_record(keyword) }
```